INFO:__main__:============================ record when total_tx_num reaches 100=====================
ERROR:__main__:Errored: 1 retry happened in running tx: PaymentTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=fe5e1299 key=/Table/54/1/4 pri=0.03483528 epo=0 ts=1635679216.632162000,0 min=1635679212.631643000,0 seq=1} lock=true stat=ABORTED rts=1635679212.631643000,0 wto=false gul=1635679213.131643000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 109, in <lambda>
    lambda l_conn: m_tx_ins.payment_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 197, in payment_transaction
    (payment_amount, c_w_id, c_d_id))
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=fe5e1299 key=/Table/54/1/4 pri=0.03483528 epo=0 ts=1635679216.632162000,0 min=1635679212.631643000,0 seq=1} lock=true stat=ABORTED rts=1635679212.631643000,0 wto=false gul=1635679213.131643000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx PaymentTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 200=====================
INFO:__main__:============================ record when total_tx_num reaches 300=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=acf9877b key=/Table/55/1/6/10 pri=0.02907873 epo=0 ts=1635680064.015081000,0 min=1635680060.013794000,0 seq=104} lock=true stat=ABORTED rts=1635680060.013794000,0 wto=false gul=1635680060.513794000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 105, in <lambda>
    lambda l_conn: m_tx_ins.new_order_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=acf9877b key=/Table/55/1/6/10 pri=0.02907873 epo=0 ts=1635680064.015081000,0 min=1635680060.013794000,0 seq=104} lock=true stat=ABORTED rts=1635680060.013794000,0 wto=false gul=1635680060.513794000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 400=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=9c7ab8ec key=/Table/55/1/6/9 pri=0.04977612 epo=0 ts=1635680306.346191000,0 min=1635680305.344779000,0 seq=98} lock=true stat=ABORTED rts=1635680305.344779000,0 wto=false gul=1635680305.844779000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 105, in <lambda>
    lambda l_conn: m_tx_ins.new_order_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=9c7ab8ec key=/Table/55/1/6/9 pri=0.04977612 epo=0 ts=1635680306.346191000,0 min=1635680305.344779000,0 seq=98} lock=true stat=ABORTED rts=1635680305.344779000,0 wto=false gul=1635680305.844779000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
ERROR:__main__:Errored: 2 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=b7fac7fb key=/Table/55/1/6/9 pri=0.04977612 epo=0 ts=1635680308.348973000,0 min=1635680307.347615000,0 seq=98} lock=true stat=ABORTED rts=1635680307.347615000,0 wto=false gul=1635680307.847615000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 105, in <lambda>
    lambda l_conn: m_tx_ins.new_order_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=b7fac7fb key=/Table/55/1/6/9 pri=0.04977612 epo=0 ts=1635680308.348973000,0 min=1635680307.347615000,0 seq=98} lock=true stat=ABORTED rts=1635680307.347615000,0 wto=false gul=1635680307.847615000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 3 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=d48c7d9c key=/Table/55/1/6/8 pri=0.04410665 epo=0 ts=1635680419.760170000,0 min=1635680416.759044000,0 seq=68} lock=true stat=ABORTED rts=1635680416.759044000,0 wto=false gul=1635680417.259044000,0 int=67
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 105, in <lambda>
    lambda l_conn: m_tx_ins.new_order_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=d48c7d9c key=/Table/55/1/6/8 pri=0.04410665 epo=0 ts=1635680419.760170000,0 min=1635680416.759044000,0 seq=68} lock=true stat=ABORTED rts=1635680416.759044000,0 wto=false gul=1635680417.259044000,0 int=67
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 500=====================
ERROR:__main__:Errored: 1 retry happened in running tx: RelCustomerTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635680569.629115000,0 encountered previous write with future timestamp 1635680569.629115000,1 within uncertainty interval `t <= (local=1635680569.636186000,0, global=1635680570.129115000,0)`; observed timestamps: [{1 1635680569.630674000,0} {4 1635680569.629115000,0} {5 1635680569.636186000,0}]: "sql txn" meta={id=b080017c pri=0.01778710 epo=0 ts=1635680569.629115000,0 min=1635680569.629115000,0 seq=0} lock=false stat=PENDING rts=1635680569.629115000,0 wto=false gul=1635680570.129115000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 142, in <lambda>
    lambda l_conn: m_tx_ins.related_customer_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 551, in related_customer_transaction
    ''', (order[0], w_id, d_id, w_id)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635680569.629115000,0 encountered previous write with future timestamp 1635680569.629115000,1 within uncertainty interval `t <= (local=1635680569.636186000,0, global=1635680570.129115000,0)`; observed timestamps: [{1 1635680569.630674000,0} {4 1635680569.629115000,0} {5 1635680569.636186000,0}]: "sql txn" meta={id=b080017c pri=0.01778710 epo=0 ts=1635680569.629115000,0 min=1635680569.629115000,0 seq=0} lock=false stat=PENDING rts=1635680569.629115000,0 wto=false gul=1635680570.129115000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
ERROR:__main__:Errored: 2 retry happened in running tx: RelCustomerTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635680578.436906000,0 encountered previous write with future timestamp 1635680578.443735000,0 within uncertainty interval `t <= (local=1635680578.444645000,0, global=1635680578.936906000,0)`; observed timestamps: [{1 1635680578.438537000,0} {4 1635680578.436906000,0} {5 1635680578.444645000,0}]: "sql txn" meta={id=8aa5fc47 pri=0.00141100 epo=0 ts=1635680578.436906000,0 min=1635680578.436906000,0 seq=0} lock=false stat=PENDING rts=1635680578.436906000,0 wto=false gul=1635680578.936906000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 142, in <lambda>
    lambda l_conn: m_tx_ins.related_customer_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 551, in related_customer_transaction
    ''', (order[0], w_id, d_id, w_id)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635680578.436906000,0 encountered previous write with future timestamp 1635680578.443735000,0 within uncertainty interval `t <= (local=1635680578.444645000,0, global=1635680578.936906000,0)`; observed timestamps: [{1 1635680578.438537000,0} {4 1635680578.436906000,0} {5 1635680578.444645000,0}]: "sql txn" meta={id=8aa5fc47 pri=0.00141100 epo=0 ts=1635680578.436906000,0 min=1635680578.436906000,0 seq=0} lock=false stat=PENDING rts=1635680578.436906000,0 wto=false gul=1635680578.936906000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
ERROR:__main__:Errored: 3 retry happened in running tx: RelCustomerTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635680587.105777000,0 encountered previous write with future timestamp 1635680587.105777000,1 within uncertainty interval `t <= (local=1635680587.113636000,0, global=1635680587.605777000,0)`; observed timestamps: [{1 1635680587.108055000,0} {4 1635680587.105777000,0} {5 1635680587.113636000,0}]: "sql txn" meta={id=e6b2b71c pri=0.00611525 epo=0 ts=1635680587.105777000,0 min=1635680587.105777000,0 seq=0} lock=false stat=PENDING rts=1635680587.105777000,0 wto=false gul=1635680587.605777000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 142, in <lambda>
    lambda l_conn: m_tx_ins.related_customer_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 551, in related_customer_transaction
    ''', (order[0], w_id, d_id, w_id)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635680587.105777000,0 encountered previous write with future timestamp 1635680587.105777000,1 within uncertainty interval `t <= (local=1635680587.113636000,0, global=1635680587.605777000,0)`; observed timestamps: [{1 1635680587.108055000,0} {4 1635680587.105777000,0} {5 1635680587.113636000,0}]: "sql txn" meta={id=e6b2b71c pri=0.00611525 epo=0 ts=1635680587.105777000,0 min=1635680587.105777000,0 seq=0} lock=false stat=PENDING rts=1635680587.105777000,0 wto=false gul=1635680587.605777000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
INFO:__main__:Running Tx RelCustomerTxParams successful at the 4 time retry
INFO:__main__:============================ record when total_tx_num reaches 600=====================
INFO:__main__:============================ record when total_tx_num reaches 700=====================
INFO:__main__:============================ record when total_tx_num reaches 800=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=2fbe0a8c key=/Table/55/1/6/10 pri=0.04336149 epo=0 ts=1635681560.793430000,0 min=1635681557.792155000,0 seq=98} lock=true stat=ABORTED rts=1635681557.792155000,0 wto=false gul=1635681558.292155000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 105, in <lambda>
    lambda l_conn: m_tx_ins.new_order_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=2fbe0a8c key=/Table/55/1/6/10 pri=0.04336149 epo=0 ts=1635681560.793430000,0 min=1635681557.792155000,0 seq=98} lock=true stat=ABORTED rts=1635681557.792155000,0 wto=false gul=1635681558.292155000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=706b776f key=/Table/55/1/6/8 pri=0.00134381 epo=0 ts=1635681595.972253000,0 min=1635681595.972253000,0 seq=104} lock=true stat=ABORTED rts=1635681595.972253000,0 wto=false gul=1635681596.472253000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 105, in <lambda>
    lambda l_conn: m_tx_ins.new_order_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=706b776f key=/Table/55/1/6/8 pri=0.00134381 epo=0 ts=1635681595.972253000,0 min=1635681595.972253000,0 seq=104} lock=true stat=ABORTED rts=1635681595.972253000,0 wto=false gul=1635681596.472253000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 900=====================
INFO:__main__:============================ using file /opt/project_files/xact_files_A/23.txt=====================
INFO:__main__:{'DeliveryTxParams': 98, 'NewOrderTxParams': 433, 'StockLevelTxParams': 11, 'TopBalanceTxParams': 9, 'PaymentTxParams': 331, 'RelCustomerTxParams': 5, 'OrderStatusTxParams': 8, 'PopItemTxParams': 6}
INFO:__main__:{'DeliveryTxParams': 98, 'NewOrderTxParams': 433, 'StockLevelTxParams': 11, 'TopBalanceTxParams': 9, 'PaymentTxParams': 331, 'RelCustomerTxParams': 5, 'OrderStatusTxParams': 8, 'PopItemTxParams': 6}
INFO:__main__:============================ total time used: 2867.742824077606 second =====================
